### PR TITLE
Test suite

### PR DESCRIPTION
**Pull request by jamiebrown201**
Adding a test suit to the template project for the search functionality as well as adding in the CAPI api request along with tests.

### CODE REVIEW

File: test/lib/fletchContent.spec.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
+const chai = require("chai");
```
Comment (Lily2point0) [inline test/lib/fletchContent.spec.js]: test file called **fletch**Content?

File: test/lib/fletchContent.spec.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+describe("lib/fetchContent", () => {
+  afterEach(() => {
+    nock.cleanAll();
+  });
+
+  describe("search", () => {
+    context("for a 200 repsonse", () => {
+      it("requests the search api with a queryString", async () => {
+        setUpSearchMock(responseBody.defualt);
```
Comment (Lily2point0) [inline test/lib/fletchContent.spec.js]: 2 occurrences of `default` misspelt

File: test/fixtures/search/responseBody.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    offset: "0",
+    aspects: ["title", "lifecycle", "location"],
+    sortOrder: "DESC",
+    sortField: "lastPublishDateTime",
+    facets: { names: ["people", "organisations", "topics"], maxElements: -1 }
+  }
+};
+
+const constraints = JSON.stringify({
+  ...defualt,
```
Comment (Lily2point0) [inline test/fixtures/search/responseBody.js]: 3 occurrences of misspelt `default`

File: test/fixtures/search/searchTerm.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const defualt = { queryString: "John Dalli" };
```
Comment (Lily2point0) [inline test/fixtures/search/searchTerm.js]: misspelt `default` times 4

**Comment by jamiebrown201** [comment]
Spell checker is now on in my ide 😬

**Comment by railsagainstignorance** [comment]
Had a play. Worked ok.
Fiddled a bit. Still works.
LGTM

**Pull request merged**